std-bits/table/src/main/java/org/enso/table/data/column/operation/cast/ToBigDecimalConverter.java

```diff
@@ -0,0 +1,124 @@
+    } else if (storage instanceof DoubleStorage doubleStorage) {
+      return convertDoubleStorage(doubleStorage, problemAggregator);
+    } else if (storage instanceof BigIntegerStorage bigIntegerStorage) {
+      return convertBigIntegerStorage(bigIntegerStorage, problemAggregator);
+    } else if (storage instanceof BoolStorage boolStorage) {
+      return convertBoolStorage(boolStorage, problemAggregator);
+    } else if (storage.getType() instanceof AnyObjectType) {
+      return castFromMixed(storage, problemAggregator);
+    } else {
+      throw new IllegalStateException(
+          "No known strategy for casting storage " + storage + " to BigDecimal.");
+    }
+  }
+
+  private Storage<BigDecimal> convertDoubleStorage(
+      DoubleStorage doubleStorage, CastProblemAggregator problemAggregator) {
+    int n = doubleStorage.size();
+    BigDecimalBuilder builder = new BigDecimalBuilder(n, problemAggregator);
+    for (int i = 0; i < n; i++) {
+      if (doubleStorage.isNothing(i)) {
+        builder.appendNulls(1);
+      } else {
+        double x = doubleStorage.getItemAsDouble(i);
+        BigDecimal bigDecimal = BigDecimal.valueOf(x);
+        builder.appendRawNoGrow(bigDecimal);
+      }
+    }
+    return builder.seal();
+  }
+
+  private Storage<BigDecimal> convertLongStorage(
+      AbstractLongStorage longStorage, CastProblemAggregator problemAggregator) {
+    int n = longStorage.size();
+    BigDecimalBuilder builder = new BigDecimalBuilder(n, problemAggregator);
+    for (int i = 0; i < n; i++) {
+      if (longStorage.isNothing(i)) {
+        builder.appendNulls(1);
+      } else {
+        long x = longStorage.getItem(i);
+        BigDecimal bigDecimal = BigDecimal.valueOf(x);
+        builder.appendRawNoGrow(bigDecimal);
+      }
+    }
+    return builder.seal();
+  }
+
+  private Storage<BigDecimal> convertBoolStorage(
+      BoolStorage boolStorage, CastProblemAggregator problemAggregator) {
+    int n = boolStorage.size();
+    BigDecimalBuilder builder = new BigDecimalBuilder(n, problemAggregator);
+    for (int i = 0; i < n; i++) {
+      if (boolStorage.isNothing(i)) {
+        builder.appendNulls(1);
+      } else {
+        boolean x = boolStorage.getItem(i);
+        BigDecimal bigDecimal = booleanAsBigDecimal(x);
+        builder.appendRawNoGrow(bigDecimal);
+      }
+    }
+    return builder.seal();
+  }
+
+  private Storage<BigDecimal> convertBigIntegerStorage(
+      BigIntegerStorage bigIntegerStorage, CastProblemAggregator problemAggregator) {
+    int n = bigIntegerStorage.size();
+    BigDecimalBuilder builder = new BigDecimalBuilder(n, problemAggregator);
+    for (int i = 0; i < n; i++) {
+      if (bigIntegerStorage.isNothing(i)) {
+        builder.appendNulls(1);
+      } else {
+        BigInteger x = bigIntegerStorage.getItem(i);
+        BigDecimal bigDecimal = new BigDecimal(x);
+        builder.appendRawNoGrow(bigDecimal);
+      }
+    }
```

Comment:
This should have a `safepoint` too

Reply:
Added to all storage converter loops. (Except `canAvoidCopying` which is all elementary operations. Or is it worth having here as well?)